ngraph/frontend/paddlepaddle/src/op/slice.cpp

```diff
@@ -0,0 +1,82 @@
+                        start_idx_node = std::make_shared<ngraph::opset6::Concat>(inputs, 0);
+                    }
+                    else
+                    {
+                        auto starts = node.get_attribute<std::vector<int32_t>>("starts");
+                        start_idx_node =
+                            opset6::Constant::create(element::i32, {starts.size()}, starts);
+                    }
+
+                    if (node.has_ng_input("EndsTensor"))
+                    {
+                        end_idx_node = node.get_ng_input("EndsTensor");
+                    }
+                    else if (node.has_ng_input("EndsTensorList"))
+                    {
+                        auto inputs = node.get_ng_inputs("EndsTensorList");
+                        end_idx_node = std::make_shared<ngraph::opset6::Concat>(inputs, 0);
+                    }
+                    else
+                    {
+                        auto ends = node.get_attribute<std::vector<int32_t>>("ends");
+                        end_idx_node = opset6::Constant::create(element::i32, {ends.size()}, ends);
+                    }
+
+                    // the shape of input, such as [1, 1, 3, 3]
+                    auto shape_node = std::make_shared<opset6::ShapeOf>(data, element::Type_t::i32);
+                    // the input dim, such as [4]
+                    auto shape_shape_node =
+                        std::make_shared<opset6::ShapeOf>(shape_node, element::i32);
+                    auto const_0_node = opset6::Constant::create(element::i32, {}, {0});
+                    auto const_max_node = opset6::Constant::create(element::i32, {}, {INT_MAX});
+                    // array [0:max)
+                    auto start_node =
+                        std::make_shared<opset6::Broadcast>(const_0_node, shape_shape_node);
+                    auto end_node =
+                        std::make_shared<opset6::Broadcast>(const_max_node, shape_shape_node);
+                    auto axes_node = opset6::Constant::create(element::i32, {axes.size(), 1}, axes);
+                    auto fixed_start_node = std::make_shared<opset6::ScatterNDUpdate>(
+                        start_node, axes_node, start_idx_node);
+                    auto fixed_end_node = std::make_shared<opset6::ScatterNDUpdate>(
+                        end_node, axes_node, end_idx_node);
+
+                    return node.default_single_output_mapping(
+                        {std::make_shared<ngraph::opset6::StridedSlice>(data,
+                                                                        fixed_start_node,
+                                                                        fixed_end_node,
+                                                                        std::vector<int64_t>{},
+                                                                        std::vector<int64_t>{})},
+                        {"Out"});
```

Comment:
could you elaborate more on why we need these start/end updates?

Reply:
@luo-cheng2021 Please explain more

Reply:
Given:
        data = [ [1, 2, 3, 4], [5, 6, 7, 8], ]   // shape is: [2, 4]
        axes = [0, 1]
        starts = [1, 0]
        ends = [2, 3]
Our process is:
1. Get 'axes': [0, 1], 'starts', 'ends'
2. Get data shape: [2,4] and dims: 2
3. Create two tensor t1 and t2, shape is the dims from step2: 2. t1: [0, 0], t2: [INT_MAX, INT_MAX]
4. Use 'ScatterNDUpdate' to update some elements in t1, the updated indexes are coming from 'axes', the contents are coming from 'starts', t1: [1, 0]; apply the similar process to t2
5. Call 'StrideSlice' with t1 and t2

Why using ScatterNDUpdate is that 'axes' may be discontinuous.

Reply:
It would be great if we had this explained in the comment somewhere in the code. Thank you